src/System.Text.Json/src/System/Text/Json/Reader/Utf8JsonReader.TryGet.cs

```diff
@@ -512,37 +512,24 @@ public bool TryGetGuid(out Guid value)
             }
 
             ReadOnlySpan<byte> span = HasValueSequence ? ValueSequence.ToArray() : ValueSpan;
-            int bytesConsumed = 0;
 
-            switch (span.Length)
+            if (_stringHasEscaping)
+            {
+                int idx = span.IndexOf(JsonConstants.BackSlash);
+                Debug.Assert(idx != -1);
+                span = JsonReaderHelper.GetUnescapedBytes(span, idx);
+            }
+
+            if (span.Length != 36 /* Valid Format D Guid length */)
             {
-                case 36:
-                    // Format D (default): nnnnnnnn-nnnn-nnnn-nnnn-nnnnnnnnnnnn
-                    return Utf8Parser.TryParse(span, out value, out bytesConsumed, 'D') && span.Length == bytesConsumed;
-                case 38:
-                    byte firstToken = span[0];
-
-                    if (firstToken == (byte)'{')
-                    {
-                        // Format B: {nnnnnnnn-nnnn-nnnn-nnnn-nnnnnnnnnnnn}
-                        return Utf8Parser.TryParse(span, out value, out bytesConsumed, 'B') && span.Length == bytesConsumed;
-                    }
-                    else if (firstToken == (byte)'(')
-                    {
-                        // Format P: (nnnnnnnn-nnnn-nnnn-nnnn-nnnnnnnnnnnn)
-                        return Utf8Parser.TryParse(span, out value, out bytesConsumed, 'P') && span.Length == bytesConsumed;
-                    }
-                    else
-                    {
-                        goto default;
-                    }
-                case 32:
-                    // Format N: nnnnnnnnnnnnnnnnnnnnnnnnnnnnnnnn
-                    return Utf8Parser.TryParse(span, out value, out bytesConsumed, 'N') && span.Length == bytesConsumed;
-                default:
-                    value = default;
-                    return false;
+                value = default;
+                return false;
             }
+
+            Debug.Assert(span.IndexOf(JsonConstants.BackSlash) == -1);
+
+            // Format D (default): nnnnnnnn-nnnn-nnnn-nnnn-nnnnnnnnnnnn
```

Comment:
Maybe this is something we should document as part of the API behavior itself (at least specifying that it only supports GUIDs with hyphens and without any decorations surrounding it).